### ISSUE TITLE
Update code generation for Kaltura's new Java SDK

### ISSUE THREAD

**Issue by zoharbabin** [kaltura/developer-platform#26]
There were updates to the Java SDK. Updates are required to the codegen.
@tan-tan-kanarek - please advise on the details of what needs to be updated.

@AnnaAleksandrowicz - please note in case we need to update any docs.

**Comment by tan-tan-kanarek** [comment]
The new lib is using executor, static services, completion callback, request builder, everything is different.

**Comment by jessp01** [comment]
Indeed. A long time ago, I already sent Bobby code samples for all the new clients, including Java.
He should be able to update the codegen templates accordingly.

**Comment by jessp01** [comment]
Found 2 issues thus far.

- In https://developer.kaltura.com:8443/workflows/Search_Discover_and_Personalize/Kaltura_Media_Library_Search;step=2

```java
        filter.setOrderBy(MediaEntryOrderBy.CREATED_AT_ASC);
```

Should be:
```java
filter.setOrderBy(MediaEntryOrderBy.CREATED_AT_DESC.getValue());
```
Since in src/main/java/com/kaltura/client/enums/MediaEntryOrderBy.java: 
```java 
public enum MediaEntryOrderBy implements EnumAsString
MediaEntryOrderBy(String value)
```


- In https://developer.kaltura.com:8443/workflows/Generate_API_Sessions/Authentication;step=1:
```java
int partnerId = "2141171";
```
Should be:
```java
int partnerId = 2141171;
```
Since partnerId is of type int. Of course, `2141171` is just for illustration purposes. Happens to be the partner ID I was logged in with...

**Comment by jessp01** [comment]
The partnerId issue is true to the other code generators as well. For languages that are not strictly typed/statically typed, the code will still work as is but it's still not right and should be fixed. For Java and C#, it will fail to compile.

**Comment by bobby-brennan** [comment]
Re: enums in Java, roger.

For the partnerId, I'm not seeing this issue - I always get ints. Is it possible that it's unique to the particular account you're logged in with (2141171)?

![image](https://user-images.githubusercontent.com/7611973/40129681-eeffacec-5902-11e8-9911-77c9a4712530.png)

**Comment by jessp01** [comment]
Hi @bobby-brennan, RE partnerId: It happened with multiple partners and in various contexts but looks like is has been resolved somewhere along the way because I no longer see it.  So that only leaves us with the Java enums issue.

**Comment by bobby-brennan** [comment]
Fixed in 9a607215edc3f62bb16bc35aa29699f469b04e08

**Comment by jessp01** [comment]
Confirmed.
